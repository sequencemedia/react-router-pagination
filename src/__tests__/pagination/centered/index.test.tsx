import React from 'react'
import renderer from 'react-test-renderer'

import {
  MemoryRouter
} from 'react-router'

import {
  toInteger,
  calculateTotalPages,
  calculatePageNumber,
  Centered
} from '#pagination/centered'

describe('react-router-pagination/pagination/centered', () => {
  describe('`toInteger`', () => {
    it('is a function', () => {
      expect(toInteger)
        .toEqual(expect.any(Function))
    })
  })

  describe('`calculateTotalPages`', () => {
    it('is a function', () => {
      expect(calculateTotalPages)
        .toEqual(expect.any(Function))
    })
  })

  describe('`calculatePageNumber`', () => {
    it('is a function', () => {
      expect(calculatePageNumber)
        .toEqual(expect.any(Function))
    })
  })

  xdescribe('<Centered />', () => {
    describe('With `pageNumber` and `totalPages`', () => {
      it('renders', () => {
        const rendered = renderer.create(
          <MemoryRouter>
            <Centered
              pageNumber={1}
              totalPages={2}
            />
          </MemoryRouter>
        )

        expect(rendered.toJSON())
          .toMatchSnapshot()
      })
    })

    describe('With `spread`', () => {
      it('renders', () => {
        const rendered = renderer.create(
          <MemoryRouter>
            <Centered
              pageNumber={5}
              totalPages={9}
              spread={3}
            />
          </MemoryRouter>
        )

        expect(rendered.toJSON())
          .toMatchSnapshot()
      })
    })
  })

  describe('`Centered.calculateTotalPages()`', () => {
    describe('Your data contains 120 items which you want to display at 10 items per page', () => {
      it('returns 12', () => {
        expect(Centered.calculateTotalPages(120, 10)).toBe(12)
      })
    })

    describe('Your data contains 60 items which you want to display at 5 items per page', () => {
      it('returns 12', () => {
        expect(Centered.calculateTotalPages(60, 5)).toBe(12)
      })
    })

    describe('Your data contains 240 items which you want to display at 20 items per page', () => {
      it('returns 12', () => {
        expect(Centered.calculateTotalPages(240, 20)).toBe(12)
      })
    })

    describe('Your data contains 121 items which you want to display at 10 items per page', () => {
      it('returns 13', () => {
        expect(Centered.calculateTotalPages(121, 10)).toBe(13)
      })
    })

    describe('Your data contains 61 items which you want to display at 5 items per page', () => {
      it('returns 13', () => {
        expect(Centered.calculateTotalPages(61, 5)).toBe(13)
      })
    })

    describe('Your data contains 241 items which you want to display at 20 items per page', () => {
      it('returns 13', () => {
        expect(Centered.calculateTotalPages(241, 20)).toBe(13)
      })
    })
  })

  describe('`Centered.calculatePageNumber()`', () => {
    describe('Min', () => {
      it('returns 1', () => {
        expect(Centered.calculatePageNumber(0, 12)).toBe(1)
      })
    })

    describe('Max', () => {
      it('returns 12', () => {
        expect(Centered.calculatePageNumber(13, 12)).toBe(12)
      })
    })
  })
})
