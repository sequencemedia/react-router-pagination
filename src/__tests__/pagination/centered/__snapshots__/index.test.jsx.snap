// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-router-pagination/pagination/centered <Centered /> With \`pageNumber\` and \`totalPages\` renders 1`] = `
<ul
  className="pagination"
>
  <li
    className="current-page"
  >
    <Link>
      <a
        data-discover="true"
        href="/1"
        onClick={[Function]}
        ref={[Function]}
      >
        <span
          className="page-number"
        />
      </a>
    </Link>
  </li>
  <li
    className="page"
  >
    <Link>
      <a
        data-discover="true"
        href="/2"
        onClick={[Function]}
        ref={[Function]}
      >
        <span
          className="page-number"
        />
      </a>
    </Link>
  </li>
</ul>
`;

exports[`react-router-pagination/pagination/centered <Centered /> With \`spread\` renders 1`] = `
<ul
  className="pagination"
>
  <li
    className="zero-page"
  >
    <Link>
      <a
        data-discover="true"
        href="/1"
        onClick={[Function]}
        ref={[Function]}
      >
        <span
          className="page-number"
        />
      </a>
    </Link>
  </li>
  <li
    className="reverse-page"
  >
    <Link>
      <a
        data-discover="true"
        href="/3"
        onClick={[Function]}
        ref={[Function]}
      >
        <span
          className="reverse"
        >
          ‹
        </span>
      </a>
    </Link>
  </li>
  <Fragment>
    <li
      className="page"
    >
      <Link>
        <a
          data-discover="true"
          href="/4"
          onClick={[Function]}
          ref={[Function]}
        >
          <span
            className="page-number"
          />
        </a>
      </Link>
    </li>
    <li
      className="current-page"
    >
      <Link>
        <a
          data-discover="true"
          href="/5"
          onClick={[Function]}
          ref={[Function]}
        >
          <span
            className="page-number"
          />
        </a>
      </Link>
    </li>
    <li
      className="page"
    >
      <Link>
        <a
          data-discover="true"
          href="/6"
          onClick={[Function]}
          ref={[Function]}
        >
          <span
            className="page-number"
          />
        </a>
      </Link>
    </li>
  </Fragment>
  <li
    className="forward-page"
  >
    <Link>
      <a
        data-discover="true"
        href="/7"
        onClick={[Function]}
        ref={[Function]}
      >
        <span
          className="forward"
        >
          ›
        </span>
      </a>
    </Link>
  </li>
  <li
    className="last-page"
  >
    <Link>
      <a
        data-discover="true"
        href="/9"
        onClick={[Function]}
        ref={[Function]}
      >
        <span
          className="page-number"
        />
      </a>
    </Link>
  </li>
</ul>
`;
